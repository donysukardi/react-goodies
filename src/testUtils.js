const propsToDataAttrs = props =>
  Object.entries(props).reduce((acc, [key, value]) => {
    acc[`data-${String(key).toLowerCase()}`] = value
    return acc
  }, {})

export {propsToDataAttrs}
