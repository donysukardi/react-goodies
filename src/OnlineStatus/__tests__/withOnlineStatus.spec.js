/* eslint-disable global-require */
import React from 'react'
import { cleanup, render } from 'react-testing-library'
import { propsToDataAttrs } from '../../testUtils'

const Target = props => (
  <div data-testid="targetContainer" {...propsToDataAttrs(props)} />
)
const dummyMapStatusToProps = () => {}

afterEach(cleanup)

describe('withOnlineStatus', () => {
  describe('Online Status API is supported', () => {
    let origOnlineStatus = null
    let withOnlineStatus = null

    afterEach(cleanup)

    beforeAll(() => {
      origOnlineStatus = global.navigator.onLine
    })

    beforeEach(() => {
      jest.resetModules()

      Object.defineProperty(global.navigator, 'onLine', {
        get: () => true,
        configurable: true,
      })
      withOnlineStatus = require('../withOnlineStatus').default
    })

    afterAll(() => {
      Object.defineProperty(global.navigator, 'onLine', {
        get: () => origOnlineStatus,
        configurable: true,
      })
    })

    describe('display name', () => {
      let origNodeEnv = null

      beforeAll(() => {
        origNodeEnv = process.env.NODE_ENV
      })

      afterAll(() => {
        process.env.NODE_ENV = origNodeEnv
      })

      it('should wrap display name in non-production env', () => {
        process.env.NODE_ENV = 'test'

        const EnhancedTarget = withOnlineStatus(dummyMapStatusToProps)(Target)
        render(<EnhancedTarget />)
        expect(EnhancedTarget.displayName).toMatchSnapshot()
      })

      it('should not wrap display name in production env', () => {
        process.env.NODE_ENV = 'production'

        const EnhancedTarget = withOnlineStatus()(Target)
        render(<EnhancedTarget />)
        expect(EnhancedTarget.displayName).toMatchSnapshot()
      })
    })
  })

  describe('Online Status API is not supported', () => {
    let origOnlineStatus = null
    let withOnlineStatus = null

    beforeAll(() => {
      origOnlineStatus = global.navigator.onLine
    })

    beforeEach(() => {
      jest.resetModules()

      Object.defineProperty(global.navigator, 'onLine', {
        get: () => undefined,
        configurable: true,
      })
      withOnlineStatus = require('../withOnlineStatus').default
    })

    afterAll(() => {
      Object.defineProperty(global.navigator, 'onLine', {
        get: () => origOnlineStatus,
        configurable: true,
      })
    })

    it('should just pass Target component through', () => {
      const EnhancedTarget = withOnlineStatus(dummyMapStatusToProps)(Target)
      const { getByTestId } = render(<EnhancedTarget a={1} b={2} c={3} />)

      expect(getByTestId('targetContainer')).toMatchSnapshot()
    })
  })
})
