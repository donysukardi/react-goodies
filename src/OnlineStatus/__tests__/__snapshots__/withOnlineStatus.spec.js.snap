// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withOnlineStatus Online Status API is not supported should just pass Target component through 1`] = `
<div
  data-a="1"
  data-b="2"
  data-c="3"
  data-testid="targetContainer"
/>
`;

exports[`withOnlineStatus Online Status API is supported display name should not wrap display name in production env 1`] = `undefined`;

exports[`withOnlineStatus Online Status API is supported display name should wrap display name in non-production env 1`] = `"withOnlineStatus(Target)"`;
